### PR TITLE
Refactor cm8jl65 driver.  Employ uniform initialization, format, and deprecate ringbuffer and IOCTL.

### PR DESCRIPTION
**Pull request by mcsauder**
**Describe problem solved by the proposed pull request**
This PR refactors the CM8JL65 driver to employ uniform initialization, move the parser and crc16 methods into the CM8JL65 class, move doxy documentation to the method declarations, and formats/standardize the driver structure.

This PR is a step in standardizing the cm8jl65 driver with other distance sensor drivers moving toward resolution of Issue #9279.

Distance sensor drivers and IMU drivers are perhaps the most numerous instances of code duplication in the drivers.  Similar work is being undertaken in the IMU drivers as well, (PR #11629).  Additional work within this driver will follow this PR.

**Additional context**
See Issue #9279.

**Test data / coverage**
This PR has been bench tested and flight tested with pixhawk 4 and pixkhawk 4 mini hardware, see comments below for data plots.

Please let me know if you have questions on this PR.  Thanks!

-Mark

### CODE REVIEW

**Comment by mcsauder** [comment]
@cmic0 , would you mind reviewing this PR?  Thanks in advance!

**Comment by mcsauder** [comment]
Hi @cmic0.  This PR gives identical results to the current master.  I will continue refactoring work with this and the other distance sensor drivers incrementally to reduce review workload.

Would you be willing/able to review/approve/reject this PR?

Here are readings from this PR
image

Here are readings from today's PX4 master branch:
image

Thank you for your review!  Let me know if you would like to see anything else from this PR!

File: src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp [hunk 1]
```diff
@@ -41,166 +41,142 @@
...
-	*/
-	void				Run() override;
+	uint16_t _crc16{0};
 
-	int				collect();
+	float _max_distance{9.0f};
+	float _min_distance{0.10f};
 
-};
+	ringbuffer::RingBuffer *_reports{nullptr};
```
Comment (dagar) [inline src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp]: Can this be gutted entirely (ringbuffer, ioctl, read, etc)?

Comment (mcsauder) [reply src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp]: Unfortunately, not that easily... See my work [here](https://github.com/mcsauder/Firmware/commit/0b9f34bf811b3947c7a324fe2db976d13097f88c). However, if we can get this PR accepted and one to follow it from the continued work in the link I think the issue will become apparent. 

I can successfully deprecate IOCTL usage from I2C devices, but not CDev devices.  For CDev devices, [these calls](https://github.com/mcsauder/Firmware/blob/0b9f34bf811b3947c7a324fe2db976d13097f88c/src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp#L636) are still required.

Comment (mcsauder) [reply src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp]: Please disregard my comments above.  I discovered the issue preventing me from succeeding in deleting the ioctl calls was that the ioctl calls were where the start() method was being called from.  Palm->forehead. :}

I have pushed up a commit that is bench tested and accomplishes your requests.  I will post a/b comparisons and flight logs as soon as I can get to that work.

**Comment by mcsauder** [comment]
Hi @cmic0 and @dagar ,  here are flight logs and plots of current PX4 master and this PR on a 250 sized drone, pixhawk 4 mini in position hold mode from ground level incrementing altitude slowly and holding at ~75cm increments up to ~3m.. The results are indistinguishable to me, so I think this PR is good to go. (I would like to squash the commits before merging to keep a clean history, please let me know when I can do this.)

PX4 master: https://review.px4.io/plot_app?log=89b6ccba-833f-454f-a0a7-f82078fbb049
image

This PR: https://review.px4.io/plot_app?log=b09b4f8a-12d4-4d1a-a062-a51852f3b189
image

**Comment by mcsauder** [comment]
This sensor is going good filtering by all accounts!

**Comment by mcsauder** [comment]
@LorenzMeier and @dagar , slowly getting this mission accomplished. :)

**Comment by mcsauder** [comment]
@dagar, ping.

**Comment by mcsauder** [comment]
@dagar and @davids5 , the cm8jl65 driver has sufficient stack allocated in PX4:master and (unchanged) with this PR:
image

**Comment by davids5** [comment]
@mcsauder - we need @dagar  confirmation but (IIRC) the 300 margin in the stackmon could be reduced to 96-100 - we only stack like 8 registers on the user process stack when an ISR is run. The rest of the penetration is application determined. Without 100% code coverage that worst case is not known. But 96 bytes of overhead is more than enough if the longest call depth has been exercised.

**Comment by mcsauder** [comment]
@davids5 and @dagar , I've reduced the cm8jl65 stack allocation from 1400 to 1200, merged latest master and rebased.

Stackcheck fails at 1100 and functions at the current commit value of 1200.

image

**Comment by davids5** [comment]
@mcsauder There are 3 things here. 

1. The stack check build - hardware test from 68+136 margin
2. The loadmon's opinion that 300 (iirc)  are needed.
3. The Actual penetration of the application + ISR overhead. 

The hard truth id the #1 will dictate 204 bytes of margin (it does not assume a separate interrupt stack)
So it is wasteful. 

So I would have stack mon warn at  216

**Comment by mcsauder** [comment]
Thank you for your review @dagar !

**Comment by mcsauder** [comment]
(and thank you for your feedback @davids5!!!)

**Pull request merged**